### ISSUE TITLE
Excluding variables from being predicted and from being predictor

### ISSUE THREAD

**Issue by LeaSchuurmans** [amices/mice#434]
So I am doing an imputation by group in mice and among others, I got two kinds of variables in my dataset, that cause trouble:

**A**s: Demographic variables I do not want to predict but still use as a preditor 
(Mainly because I don't need them for my analyses but they prolong the time that the imputation needs.)

**B**s: Variables I want to be predicted but don't want to use as a predictor

Background for Bs: They are sumscores that cause the following logged events:
`   it im                  dep    meth                           out`
`   1  1                   C       bygroup                     B`

So I am specifying my predMatrix with quickpred:
`predMat = quickpred(data, mincor = 0.05)`
`predMat[c("A1", "A2), ] = 0` 
`predMat[, c("B1", "B2)] = 0` 

` missvars = colnames(data)[colSums(is.na(data)) > 0]` 
` catvars = c("A1", "A2", "A3", "A5")`

` method = as.character(rep("", ncol(data))) %>% ` 
`   {names(.) = colnames(data);.}` 
` method[missvars] = "bygroup"` 

` impFun = as.list(rep("", ncol(data))) %>% ` 
`   {names(.) = colnames(data);.}` 
` impFun[missvars] = "pmm"`

`data%>% `
  `select(catvars) %>% `
`  map_dfr(~as.numeric(as.character(.))) %>% `
 ` cbind(data%>% select(-catvars), .) -> data`

However, when I start my imputation with mice:
`imp.test = mice::mice(data, method = method,
                      predictorMatrix = predMat,
                      group = "group", seed = 123,
                      imputationFunction = impFun,
                      m = 5, maxit = 5)`

The console displays, that mice does indeed still impute the As and the logged Events concerning the Bs also remain, which I think means that they are still used a predictors.

I tried a more simple imputation without bygroup but the As remained to be imputed, and even though the logged events for Bs are not displayed, I think they are still used as predictors.

Is there another way to adjust this instead of changing the predictorMatrix?

I thank you in advance for any help and your time.

**Comment by stefvanbuuren** [comment]
Q1: "A: Variables I do not want to impute but still use as a predictor"

Two possibilities:
- Specify "" as you imputation `method`. 
- Skip the variable in `visitSequence`.

Be aware that using these variables as predictor for other variables will propagate the `NA`'s, so this is not generally recommended. See documentation. Your solution using `predictorMatrix` does not work because these are still imputed using a simple intercept model.

Q2: "Variables to be imputed but don't want to use as a predictor"

Your approach with `predictorMatrix` just does that.

**Comment by LeaSchuurmans** [comment]
@stefvanbuuren thank you really much for your answere.
I removed the visit sequence.

The logged events concerning the Bs disappeared when I removed this part:
```
data%>%
select(catvars) %>%
map_dfr(~as.numeric(as.character(.))) %>%
cbind(data%>% select(-catvars), .) -> data
```
I changed the method accordingly to "" and indeed this resulted in some other variables not being fully imputed, so I now understand why it is not recommended. 
So it seems easier to accept the longer computing time and impute the variables anyway, even if they are not needed for the later analyses.
